chore: Various improvements spotted in the project

banjoh opened the pull request:
* Run linter in CI - this was earlier missed
* Run trivy scanner in CI
* Remove deprecated functions
* Use gorilla's logging middleware
* Bind server port to localhost interface
* Fix linter errors

For logging, here is a before and after log output

**Before**
```
INFO[0040] 200 GET /apis/storage.k8s.io/v1?timeout=32s
INFO[0040] 200 GET /apis/certificates.k8s.io/v1?timeout=32s
INFO[0040] 200 GET /apis/admissionregistration.k8s.io/v1beta1?timeout=32s
INFO[0040] called getAPIV1NamespaceResource
INFO[0040] 200 GET /api/v1/namespaces/projectcontour/pods/envoy-27r82
INFO[0040] called getAPIV1NamespaceResourceLog
2023/10/06 11:18:08 http: superfluous response.WriteHeader call from github.com/replicatedhq/sbctl/pkg/api.(*logRecord).WriteHeader (middleware.go:20)
INFO[0040] 200 GET /api/v1/namespaces/projectcontour/pods/envoy-27r82/log?container=shutdown-manager
```

**After**: Note that we now have payload size and query parameters
```
127.0.0.1 - - [06/Oct/2023:11:25:26 +0100] "GET /apis/coordination.k8s.io/v1beta1?timeout=32s HTTP/1.1" 200 293
127.0.0.1 - - [06/Oct/2023:11:25:26 +0100] "GET /apis/batch/v1beta1?timeout=32s HTTP/1.1" 200 437
127.0.0.1 - - [06/Oct/2023:11:25:26 +0100] "GET /apis/rbac.authorization.k8s.io/v1?timeout=32s HTTP/1.1" 200 914
127.0.0.1 - - [06/Oct/2023:11:25:26 +0100] "GET /apis/rbac.authorization.k8s.io/v1beta1?timeout=32s HTTP/1.1" 200 919
INFO[0023] called getAPIV1NamespaceResource
127.0.0.1 - - [06/Oct/2023:11:25:26 +0100] "GET /api/v1/namespaces/projectcontour/pods/envoy-27r82 HTTP/1.1" 200 10910
INFO[0023] called getAPIV1NamespaceResourceLog
2023/10/06 11:25:26 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
127.0.0.1 - - [06/Oct/2023:11:25:26 +0100] "GET /api/v1/namespaces/projectcontour/pods/envoy-27r82/log?container=shutdown-manager HTTP/1.1" 200 102
```

> Couldn't we have added the query parameters to the current logging middleware implementation?

Yes, we could have, but why reimplement the wheel?

CpuID reviewed (changes requested):
Was going to approve (code looks fine), but decided to build it and test out the binary locally also.

Noticed a substantial increase in these messages prefixing the output of commands... not sure what we can do about those (I've seen 1-2 lines of it at times when trying to run `kubectl logs` in the past, but this is a lot more, and for `get po` too). I've trimmed the output, there was probably 5x what I pasted below.

```
~/git/banjoh/sbctl em/various-improvements ❯ kubectl get po -A                                                                                                                             ⎈ default 09:09:17
2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                               2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                 2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                                                2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                 2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                  2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                                                                 2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                  2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                   2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                                                                                  2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                   2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                    2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
     2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                                    2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                     2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                      2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                            
...

                                                                                                                                                                                                2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                                                                                 2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
                                                                  2023/10/09 09:09:19 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:61)
   NAMESPACE     NAME                                                         READY   STATUS                       RESTARTS        AGE
...<the pod output as normal after>
```

banjoh commented:
> Noticed a substantial increase in these messages prefixing the output of commands... 

This was a good catch. Thanks for also running the application as part of the review.

This was a side-effect of the logging changes. Its now fixed. 

The gist of the matter was that the ["superfluous response write"](https://github.com/golang/go/blob/2744155d369ca838be57d1eba90c3c6bfc4a3b30/src/net/http/server.go#L1151) is logged when a program tries to write a header after the body has been written to a response. Its not allowed unless the header is a [trailer header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer). I have fixed both places where this happens in [this commit](https://github.com/replicatedhq/sbctl/pull/126/commits/eff97c435d5ed295bf641f2df3c6deb2f4306dbc). The code was there already, we just go hit by the warning after making this change. I have also made sure (in the same commit) that error logs, such as this one, logged by golang's server are logged using logrus instead of using the standard logger.

CpuID reviewed (approved):
Much better now with that latest commit 👍 thanks

DexterYan commented on pkg/api/server.go:
```diff
@@ -1175,6 +1186,7 @@ func filterObjectsByFields(object runtime.Object, selector fields.Selector) runt
 	case *corev1.EventList:
 		filtered := &corev1.EventList{}
 		for _, item := range o.Items {
+			item := item
```
Is it necessary since there is no goroutine inside the loop?

banjoh replied:
> Is it necessary since there is no goroutine inside the loop?

With this specific code, its not necessary. Just being defensive in case `eventToSelectableFields(&item)` function were to try to store the `&item` variable.

Also, the for-loop variable scope behaviour is not limited to goroutines. It occurs whenever a loop variable pointer gets referenced outside or the for loop scope

```go
package main

func main() {
	s := []string{"a", "b", "c"}
	r := []*string{}

	for _, v := range s {
		// v := v for-loop var scope workaround
		r = append(r, &v)
	}

	for _, v := range r {
		println(*v)
	}
}
```

Thankfully, we won't need this workaround in [go 1.22 onwards](https://github.com/golang/go/wiki/LoopvarExperiment). Its experimental at the moment in go 1.21 (`GOEXPERIMENT=loopvar go run main.go`)
